src/Native/Unix/System.Native/pal_networking.cpp

```diff
@@ -2134,38 +2134,37 @@ SystemNative_SetSockOpt(intptr_t socket, int32_t socketOptionLevel, int32_t sock
     //
     if (socketOptionLevel == PAL_SOL_SOCKET)
     {
-        if (socketOptionName == PAL_SO_EXCLUSIVEADDRUSE)
+        // Windows supports 3 address sharing modes:
+        // - not sharing      (SO_EXCLUSIVEADDRUSE=1, SO_REUSEADDR=0)
+        // - explicit sharing (SO_EXCLUSIVEADDRUSE=0, SO_REUSEADDR=1)
+        // - implicit sharing (SO_EXCLUSIVEADDRUSE=0, SO_REUSEADDR=0)
+        // On Unix we have two address sharing modes:
+        // - not sharing      (SO_REUSEPORT=0)
+        // - explicit sharing (SO_REUSEPORT=1)
+        // We make both PAL_SO_REUSEADDR and PAL_SO_EXCLUSIVEADDRUSE control SO_REUSEPORT.
+        if (socketOptionName == PAL_SO_EXCLUSIVEADDRUSE || socketOptionName == PAL_SO_REUSEADDR)
```

Comment:
If we're now allowing ExclusiveAddressUse to be set, should we also update the multi-connect support to track it? e.g.
https://github.com/dotnet/corefx/blob/969d86808c24fae13ab15c636b41ebc4ed204534/src/System.Net.Sockets/src/System/Net/Sockets/Socket.Unix.cs#L35

Reply:
Yes we should. I'll add that.

Reply:
Thinking about this a bit more. Connects don't fail for UDP. And for TCP, the ExclusiveAddress/ReuseAddress is an option for listen sockets (which don't connect). So there is no compelling use-case to add this to the tracked socket options.

Reply:
Ok.